### DISCUSSION TITLE
在宝塔面板上部署DailyHotApi

### DISCUSSION THREAD

**Discussion by dahaichaoyong** [imsyy/DailyHotApi#27] [General]
非常感谢作者@imsyy 开发的系统，不过在评论区看到有小伙伴不大明白怎么部署，今天忙活了半天，也遇到了“-bash: pnpm: command not found”、“-bash: /usr/bin/pnpm: Permission denied”这类问题，最后终于部署成功了，所以在这里分享下我部署DailyHotApi的步骤。（我也是个小白，如有错漏，请谅解）（因为没看到图片上传按钮，所以都是文字了） 

1、解析域名 [hot-api.seoshipin.cn]
2、在宝塔网站管理-创建站点，网站PHP版本选择【纯静态】。
3、把源码上传到网站根目录。
4、（接下来，重点来了）安装pnpm
（1）我们先到宝塔的软件管理，搜索pm2，安装【PM2管理器5.5】 
（2）点击PM2管理器 5.5右边的设置，点击Node版本，我选择了v16.20.2，等待切换完成
（3）点击【模块管理】，在搜索框输入pnpm，点击【安装】
（4）回到网站管理，进入网站根目录，点击【终端 】
然后在终端内依次输入：
pnpm install
pnpm start

说明：
pnpm install 是安装依赖
pnpm start 是运行

Tips：
服务端口以及允许的域名可在根目录下的.evn文件内修改

（5）终端跑完后前往站点设置 – 反向代理 – 添加反向代理
代理名称：api-hot
目标url：http://127.0.0.1:6688
发送域名：$host
其他不用管，点击提交

（6）最后前往 软件商店 – PM2管理器 – 添加项目
启动文件：选择网站根目录的index.js，比如我的：/www/wwwroot/hot-api.seoshipin.cn/index.js
其他的默认不用管，点击提交

（7）温馨提醒：如果使用的是如 腾讯云 阿里云 华为云等服务器，需要前往安全组/防火墙 添加设置的服务器端口6688，其它服务器需要在宝塔面板 – 安全 – 添加端口规则

至此，API就部署好了。

------------------------------------------------------------------
接下来去部署前端页面，方法是类似的。
1、解析域名 [hot.seoshipin.cn]
2、在宝塔网站管理-创建站点，网站PHP版本选择【纯静态】。
3、把源码上传到网站根目录。
Tips：
如果你需要修改成你自己部署的API接口
请修改根目录下的 .evn 文件内的 VITE_GLOBAL_API

4、进入网站根目录，点击【终端 】
然后在终端内依次输入：
pnpm install
pnpm build

5、构建完成后可将生成的 **dist** 文件夹内的文件移动至站点根目录即可

**Comment by dahaichaoyong** [comment]
希望可以增加 小红书的话题排行榜

**Reply by fhbxfhh** [reply]
PHP版本选择伪静态？找不到啊，只找到纯静态！！！

**Reply by fhbxfhh** [reply]
回到网站管理，进入网站根目录，点击【终端 】
这个我是真没搞懂，根目录没有终端，，，mp2启动提示[PM2][ERROR] File ecosystem.config.js not found

**Reply by imsyy** [reply]
是在项目根目录中开启终端 @fhbxfhh 
![image](https://github.com/imsyy/DailyHotApi/assets/42232682/09d54232-8ae5-486d-8eb5-5e68df0e3c2c)

**Reply by fhbxfhh** [reply]
win系统2012宝塔面板，MP2启动不了，太难了

**Reply by fhbxfhh** [reply]
![image](https://github.com/imsyy/DailyHotApi/assets/86190990/52c1919d-aa6a-4662-ba0d-a3d386c04a7d)

**Reply by imsyy** [reply]
用这个试试
![image](https://github.com/imsyy/DailyHotApi/assets/42232682/156210db-1120-4f4a-9af2-56c4f32f919c)

**Reply by fhbxfhh** [reply]
我宝塔面板没有这个，算了，不弄了，太难了。

**Reply by dahaichaoyong** [reply]
打错字，是 纯静态。已修改。

**Reply by James904141** [reply]
pnpm build构建没有出现dist文件夹怎么办，搭建了好几遍都是这个情况，求解决
![QQ拼音截图20231212181213](https://github.com/imsyy/DailyHotApi/assets/145677216/bf84f48b-574b-451e-8219-bd0b428e0061)
![QQ拼音截图20231212184950](https://github.com/imsyy/DailyHotApi/assets/145677216/33a178ec-630a-4378-97db-9d6cb46e8755)

**Comment by fhbxfhh** [comment]
感谢老表

**Comment by imsyy** [comment]
> PHP版本选择伪静态？找不到啊，只找到纯静态！！！

纯静态在这里
![image](https://github.com/imsyy/DailyHotApi/assets/42232682/11691ed1-f36c-4a73-a63a-9b3cca32b4c3)

**Reply by imsyy** [reply]
检查你的 env 文件配置，“FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory” 这个是你的服务器内存满了，可以在空闲时再试下

**Comment by James904141** [comment]
大佬，没有生成dist文件夹呀

**Reply by imsyy** [reply]
你部署错项目了

**Comment by Dnyo666** [comment]
docker一键部署api不香吗，反而是页面我用的vercel

**Comment by RuSheng920** [comment]
![{3CAC6C1B-F479-4dbf-8B3E-706D33860029}](https://github.com/imsyy/DailyHotApi/assets/121843951/86c14e18-1e84-4333-bc72-60da919789c6)
啊，为啥

**Reply by yaowangmx** [reply]
请问这个如何一键部署呢？

**Comment by RuSheng920** [comment]
![{7E6C5182-FABA-4289-B369-89ADB994A54B}](https://github.com/imsyy/DailyHotApi/assets/121843951/2e0c8ece-bf68-4eea-8c69-0a4241cf1aed)
都16.20.2为什么我的显示这个

**Comment by imsyy** [comment]
> ![{7E6C5182-FABA-4289-B369-89ADB994A54B}](https://private-user-images.githubusercontent.com/121843951/328469673-2e0c8ece-bf68-4eea-8c69-0a4241cf1aed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwNzQ4NjEsIm5iZiI6MTcxNTA3NDU2MSwicGF0aCI6Ii8xMjE4NDM5NTEvMzI4NDY5NjczLTJlMGM4ZWNlLWJmNjgtNGVlYS04YzY5LTBhNDI0MWNmMWFlZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QwOTM2MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02M2FhN2EzM2I5M2MxYmQyYjA1NmUxYjJlN2ZiMmJhMjkyY2MwYTFiYWE0NGE4YjY2MDNiNjIxMmNjOGIyNmE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.-eb_0kFhsfG7urIInAXWyczReK63SqCv04BlgGbgPr0) 都16.20.2为什么我的显示这个

目前需要 node 18 及以上版本

**Reply by RuSheng920** [reply]
好的，我看楼主是16.20
